accounts/abi/bind/backends/simulated.go

```
@@ -527,9 +551,30 @@ func (b *SimulatedBackend) EstimateGas(ctx context.Context, call ethereum.CallMs
 // state is modified during execution, make sure to copy it if necessary.
 func (b *SimulatedBackend) callContract(ctx context.Context, call ethereum.CallMsg, block *types.Block, stateDB *state.StateDB) (*core.ExecutionResult, error) {
 	// Ensure message is initialized properly.
-	if call.GasPrice == nil {
+	// EIP1559 guards
+	// If we have finalized EIP1559 and do not have a properly formed EIP1559 trx, sub in default values
+	eip1559 := b.config.IsEIP1559(block.Number())
+	eip1559Finalized := b.config.IsEIP1559Finalized(block.Number())
+	if eip1559Finalized && (call.MaxMinerBribePerGas == nil || call.FeeCapPerGas == nil || call.GasPrice != nil) {
+		call.MaxMinerBribePerGas = big.NewInt(1)
+		call.FeeCapPerGas = big.NewInt(10)
+		call.GasPrice = nil
+	}
+	// If we have not activated EIP1559 and do not have a properly formed legacy trx, sub in default values
+	if !eip1559 && (call.MaxMinerBribePerGas != nil || call.FeeCapPerGas != nil || call.GasPrice == nil) {
```

Would it be better to throw an error in that case ?